query-engine/connector-test-kit-rs/query-tests-setup/src/connector_tag/js/external_process.rs

```diff
@@ -81,15 +97,34 @@ impl ExecutorProcess {
     }
 }
 
-pub(super) static EXTERNAL_PROCESS: Lazy<ExecutorProcess> =
-    Lazy::new(|| match std::thread::spawn(ExecutorProcess::new).join() {
-        Ok(Ok(process)) => process,
-        Ok(Err(err)) => exit_with_message(1, &format!("Failed to start node process. Details: {err}")),
-        Err(err) => {
-            let err = err.downcast_ref::<String>().map(ToOwned::to_owned).unwrap_or_default();
-            exit_with_message(1, &format!("Panic while trying to start node process.\nDetails: {err}"))
+/// Wraps an ExecutorProcess allowing for restarting it.
+///
+/// A node process can die for a number of reasons, being one that any `panic!` occurring in Rust
+/// asynchronous code are translated to an abort trap by wasm-bindgen, which kills the node process.
+#[derive(Clone)]
+pub(crate) struct RestartableExecutorProcess {
+    p: Arc<Mutex<ExecutorProcess>>,
```

Comment:
Also, could this be `RwLock` rather than `Mutex`?

Reply:
RwLock is not Sync, let me think how I can use it from different threads obtaining multiple references. (I might not be able to) because we need to mutate the restaratable executor process.